src/commandProcessExecution.ts

```diff
@@ -6,14 +6,29 @@ import { EnvironmentVariables } from "./domain";
 
 @provide(CommandProcessExecutionFactory)
 export class CommandProcessExecutionFactory {
-  createCommandProcessExecution(
-    command: string,
-    args?: string[],
-    cwd?: string,
-    token?: CancellationToken,
-    envVars?: EnvironmentVariables,
-  ) {
-    return new CommandProcessExecution(command, args, cwd, token, envVars);
+  createCommandProcessExecution({
+    command,
+    args,
+    stdin,
+    cwd,
+    token,
+    envVars,
+  }: {
+    command: string;
+    args?: string[];
+    stdin?: string;
+    cwd?: string;
+    token?: CancellationToken;
+    envVars?: EnvironmentVariables;
+  }) {
```

Comment:
Refactored here to get those arguments as object since most of them are optional.